include/sound/soc-acpi-intel-match.h

```diff
@@ -47,4 +47,7 @@ extern struct snd_soc_acpi_mach snd_soc_acpi_intel_mtl_sdw_machines[];
  */
 extern struct snd_soc_acpi_mach snd_soc_acpi_intel_hda_machines[];
 
+/* Intel version of snd_soc_acpi_codec_list function */
+struct snd_soc_acpi_mach *snd_soc_acpi_intel_codec_list(void *arg);
```

Comment:
your commit message also needs to add a lot more background on what you are trying to solve, and how. The backwards-compatibility story needs to be crystal-clear.